src/gridcoin/voting/registry.cpp

```diff
@@ -303,12 +309,276 @@ int64_t PollReference::Age(const int64_t now) const
+        arith_uint256 target;
+        target.SetCompact(pindex->nBits);
+
+        arith_uint256 net_weight = ~arith_uint256() / arith_uint256(450) / target * arith_uint256(COIN);
+
+        arith_uint256 money_supply = pindex->nMoneySupply;
+
+        tally_active_vote_weight(net_weight, money_supply, scaled_pool_magnitude, scaled_network_magnitude);
+
+        // If voting logging category is active, log the first block and every superblock
+        if (blocks == 1 || pindex->IsSuperblock()) {
+            LogPrint(BCLog::LogFlags::VOTE, "INFO: %s: tally_active_vote_weight: net_weight = %f, money_supply = %f, "
+                                            "pool_magnitude = %f, network_magnitude = %f, block height = %i, "
+                                            "blocks = %u, active_vote_weight_tally = %f, active_vote_weight = %f.",
+                     __func__,
+                     net_weight.getdouble() / (double) COIN,
+                     money_supply.getdouble() / (double) COIN,
+                     scaled_pool_magnitude.getdouble() / 100.0,
+                     scaled_network_magnitude.getdouble() / 100.0,
+                     pindex->nHeight,
+                     blocks,
+                     active_vote_weight_tally.getdouble() / (double) COIN,
+                     (active_vote_weight_tally / arith_uint256(blocks)).getdouble() / (double) COIN
+                     );
+        }
+
+        // Log the last block and break if at pindex_end.
+        if (pindex == pindex_end) {
+            LogPrint(BCLog::LogFlags::VOTE, "INFO: %s: tally_active_vote_weight: net_weight = %f, money_supply = %f, "
+                                            "pool_magnitude = %f, network_magnitude = %f, block height = %i, "
+                                            "blocks = %u, active_vote_weight_tally = %f, active_vote_weight = %f.",
+                     __func__,
+                     net_weight.getdouble() / (double) COIN,
+                     money_supply.getdouble() / (double) COIN,
+                     scaled_pool_magnitude.getdouble() / 100.0,
+                     scaled_network_magnitude.getdouble() / 100.0,
+                     pindex->nHeight,
+                     blocks,
+                     active_vote_weight_tally.getdouble() / (double) COIN,
+                     (active_vote_weight_tally / arith_uint256(blocks)).getdouble() / (double) COIN
+                     );
+
+            break;
+        }
+    }
+
+    if (!blocks) {
+        LogPrintf("WARNING: %s: No blocks tallied for poll txid %s, title \"%s\", so no active vote weight can be "
+                  "calculated.",
+                 __func__,
+                 Txid().GetHex(),
+                 Title()
+                 );
+
+        return std::nullopt;
+    }
+
+    arith_uint256 active_vote_weight = active_vote_weight_tally / arith_uint256(blocks);
+
+    g_timer.GetTimes("finished execution for poll id " + Txid().GetHex() + " title " + Title(), __func__);
```

Comment:
Are we interested in seeing this even if no blocks were found?

Reply:
You are talking about the timer output?

Reply:
I think you are, and I can put a conditional around it.

Reply:
No, you can't be... it will return before that, because of the divide by zero protection. You must be talking about the no blocks log warning above. Here I think it should be logged. Since a poll should have at least one block, it would be very unnatural for blocks to be zero, so the divide by zero protection should be expected never to be activated. Given that, if it is encountered, at least a warning log entry should be put in the log. Quite frankly, it might even should be an error.

Reply:
Aye, I would probably move it to above the conditional return. This is me assuming that `GetTimes` will log.

Reply:
Quite frankly, looking at the above code I wrote very carefully, blocks=0 is impossible. If pindex_start is a nullptr, the function immediately returns std::nullopt. If pindex_end is nullptr, The function assigns pindexBest to pindex_end. The net of that is that there will be at least one valid pindex to iterate in the for loop where the avw tally is executed and the block counter is incremented. So I think I can remove the divide by zero blocks protection entirely and also the log entry for that. Do you agree?

Reply:
What I meant was that regardless of if we found a block or not, it would possibly be beneficial to print the time it took to reach that conclusion :)
But I think you might be right regarding the preconditions. It should also be fairly easy to test with a unit test.

Reply:
Eh... I went ahead and added the log entry.